### DISCUSSION TITLE
Oneplus 8T phone connects, but no internet

### DISCUSSION THREAD

**Discussion by ffxxcad** [srsran/srsRAN_Project#135] [General Help] [answered]
**Files:**
[amf_log.txt](https://github.com/srsran/srsRAN_Project/files/12100753/amf_log.txt)
[gNB_log.txt](https://github.com/srsran/srsRAN_Project/files/12100754/gNB_log.txt)
[gNB_mac.zip](https://github.com/srsran/srsRAN_Project/files/12100761/gNB_mac.zip)
[gNB_ngap.zip](https://github.com/srsran/srsRAN_Project/files/12100764/gNB_ngap.zip)

**Setup:**
- Ettus USRP B210
- National Instruments CDA 2990 
- Oneplus 8T (European version)
- Gialer Programable Sim Card
- Open5GS Core

The gNB, amf, and upf configuration files are identical to the ones found [here](https://docs.srsran.com/projects/project/en/latest/tutorials/source/cotsUE/source/index.html#configuration).

**Issue:**
Phone appears to be authenticated by the core but does not get a PDU Session and consequently does not have internet. We have tried these steps [here](https://docs.srsran.com/projects/project/en/latest/tutorials/source/cotsUE/source/index.html#no-internet-access) without success.

Any thoughts? Thanks in advance.

**Comment by brendan-mcauliffe** [comment]
Hi @ffxxcad, 

Typically, when the phone sees the network but isn't assigned a PDU, there is some misconfiguration between the core and the phone. I would check the following: 

- Check that the APN configuration is _exactly_ the same between the core and phone, also make sure it is set to IPv4 only. 
- Make sure that VoLTE/ VoNR is disabled. 

If this issue continues, please send the logs and pcaps for the latest test.

**Reply by ffxxcad** [reply]
Thanks for your reply. I double checked the APN in the core and the phone; both match. I also made sure that it was set to IPv4 only. VoLTE is disabled as well.

**Logs from latest test:**
[amf_log.txt](https://github.com/srsran/srsRAN_Project/files/12111902/amf_log.txt)
[gNB_log.txt](https://github.com/srsran/srsRAN_Project/files/12111903/gNB_log.txt)
[UE_radio_log.txt](https://github.com/srsran/srsRAN_Project/files/12111904/UE_radio_log.txt)
[gNB_mac.zip](https://github.com/srsran/srsRAN_Project/files/12111906/gNB_mac.zip)
[gNB_ngap.zip](https://github.com/srsran/srsRAN_Project/files/12111907/gNB_ngap.zip)

**Screenshots:**
Phone settings:
![Phone_Settings](https://github.com/srsran/srsRAN_Project/assets/132515083/6d732d2e-9704-4800-bf59-e30f40513127)

Open5GS Subscriber Information:
![Open5GS_Sub_Info](https://github.com/srsran/srsRAN_Project/assets/132515083/1ff48be6-d9c8-4cd3-a285-226e4b31de59)

**Reply by herlesupreeth** [reply]
I see in your APN setting on the phone that `APN type` is set to `ims` rather it should be set to `default`

**Reply by ffxxcad** [reply]
@herlesupreeth I changed the `APN type` to `default` and saved the access point. Upon opening the Access point again I noticed that `APN type` has been reset to `ims`, leading me to believe that `ims` is the default parameter.

**Reply by herlesupreeth** [reply, marked as answer]
I have seen this behavior when the phone has a specific APN configuration baked into the OS which gets loaded as soon as you insert a SIM of particular PLMN. You can try the following method

Method 1:

1. Go to APN settings, then from the menu select --> "Reset APN to default"
2. The above will re-populate the APN stored in the OS for a PLMN
3. Add a new APN (`srsapn` with APN type as `default`)
4. Delete all other APN and restart the phone
5. Then, give it a try.

Method 2:

1. Same as above (i.e. reset APN to default, add new APN and delete all others)
2. For the `srsapn`, set the APN type as `default,ims,ia`
3. Then, give it a try

If both the method fails, then most probably you would have to root your phone and change the APN XML file in the OS.

**Reply by ffxxcad** [reply]
@herlesupreeth, thank you so much! 

Method 1 helped the phone establish a PDU session, and in turn, an IP address (which previously did not work). However, the phone still did not have internet but could ping and iperf back and forth to the gNB. 

Internet connection was established by re-running the commands listed under the "Adding a route for the UE to have WAN connectivity" section of the [Open5GS quickstart guide](https://open5gs.org/open5gs/docs/guide/01-quickstart/). 

Both iperf and internet connection is now working on the phone. Thanks again!
